packages/transport/src/modules/relationshipTemplates/RelationshipTemplateController.ts

```diff
@@ -124,25 +144,43 @@ export class RelationshipTemplateController extends TransportController {
 
     public async fetchCaches(ids: CoreId[]): Promise<{ id: CoreId; cache: CachedRelationshipTemplate }[]> {
         if (ids.length === 0) return [];
+        const templates = await this.readRelationshipTemplates(ids.map((id) => id.toString()));
+
+        const backboneRelationshipTemplates = await (
+            await this.client.getRelationshipTemplates({
+                templates: await Promise.all(
+                    templates.map(async (t) => {
+                        const password = t.password ?? t.pin;
+                        return { id: t.id.toString(), password: password ? (await CoreCrypto.deriveHashOutOfPassword(password, t.salt!)).toBase64() : undefined };
+                    })
+                )
+            })
+        ).value.collect();
 
-        const backboneRelationships = await (await this.client.getRelationshipTemplates({ ids: ids.map((id) => id.id) })).value.collect();
+        const decryptionPromises = backboneRelationshipTemplates.map(async (t) => {
+            const template = templates.find((template) => template.id.toString() === t.id);
+            if (!template) return;
+            return { id: CoreId.from(t.id), cache: await this.decryptRelationshipTemplate(t, template.secretKey) };
+        });
 
-        const decryptionPromises = backboneRelationships.map(async (t) => {
-            const templateDoc = await this.templates.read(t.id);
+        const caches = await Promise.all(decryptionPromises);
+        return caches.filter((c) => c !== undefined);
+    }
+
+    private async readRelationshipTemplates(ids: string[]): Promise<RelationshipTemplate[]> {
+        const templatePromises = ids.map(async (id) => {
+            const templateDoc = await this.templates.read(id);
             if (!templateDoc) {
                 this._log.error(
-                    `Template '${t.id}' not found in local database and the cache fetching was therefore skipped. This should not happen and might be a bug in the application logic.`
+                    `Template '${id}' not found in local database. This should not happen and might be a bug in the application logic. Skipping further processing of that Template.`
```

Comment:
you introduced a function called `readRelationshipTemplates` in which body you log an error that if not found `Skipping further processing of that Template`. **Nothing** in the method naming indicates that there is some kind of processing that could be skipped.

Reply:
I'll just remove that part from the log then - otherwise I'd have to return the failed templates and then handle them, which seems too much effort for something that shouldn't happen

Reply:
great.